libcodechecker/cmd/product_client.py

```diff
@@ -33,16 +35,37 @@ def handle_list_products(args):
             results.append({product.endpoint: product})
         print(CmdLineOutputEncoder().encode(results))
     else:  # plaintext, csv
-        header = ['Status', 'Endpoint', 'Name', 'Description']
+        header = ['Database status', 'Endpoint', 'Name', 'Description']
         rows = []
         for product in products:
             name = base64.b64decode(product.displayedName_b64) \
                 if product.displayedName_b64 else ''
             description = base64.b64decode(product.description_b64) \
                 if product.description_b64 else ''
-            rows.append(('Database error' if not product.connected
-                         else 'No access' if not product.accessible
-                         else '',
+
+            db_status_msg = 'OK'
+            db_status = product.databaseStatus
+            if db_status == DBStatus.MISSING:
+                db_status_msg = 'Missing database.'
+            elif db_status == DBStatus.FAILED_TO_CONNECT:
+                db_status_msg = 'Connection failed.'
+            elif db_status == DBStatus.SCHEMA_MISMATCH_OK_MIGRATION:
+                db_status_msg = \
+                    'Database schema mismatch migration is available.'
+            elif db_status == DBStatus.SCHEMA_MISMATCH_NO_MIGRATION:
+                db_status_msg = \
+                    'Database schema mismatch migration is not available.'
+            elif db_status == DBStatus.SCHEMA_MISSING:
+                db_status_msg = 'Database schema is missing.'
+            elif db_status == DBStatus.SCHEMA_INITIALIZATION_ERROR:
+                db_status_msg = 'Schema initialization error.'
+            elif db_status == DBStat us.SCHEMA_UPGRADE_FAILED:
+                db_status_msg = 'Schema upgrade failed.'
+
+            if db_status == DBStatus.OK and not product.accessible:
```

Comment:
The header says `Database status` yet this isn't a *database* status. Also... is the database error or the users' no access more "powerful" or the no access?

I think: if the user has no access to a product (which information is always available in the configuration database if the server is alive), should they see database statuses?